store/listings/models.py

```diff
@@ -1,12 +1,26 @@
+from django.contrib.postgres.fields import ArrayField
+from django.core.exceptions import ValidationError
 from django.db import models
 
+from store.common.fields import PriceField
+from store.common.models import ProjectModel
 
-class Listing(models.Model):
-    listing_id = models.CharField(max_length=255, unique=True)
+
+def validate_currency(value):
+    if value != 'EUR':
```

Comment:
It seems API is sending currency code values in lower cases, make sure all the places in your code are expecting lowercase "EUR".

Change `default='EUR'` in line 23 to `default='eur'`

<img width="244" alt="Screenshot 2024-07-23 at 11 38 11" src="https://github.com/user-attachments/assets/c7ade792-ff1d-442e-a2e2-904d9f20d2fb">